### ISSUE TITLE
[Steam Deck][Linux] Crash when switching audio output from system

### ISSUE THREAD

**Issue by cmalvi** [RPCS3/rpcs3#12798]
### Quick summary

Running rpcs3 from gamemode's Steam Deck, connecting my bluetooth headphones and switch the output to them, then the emulator crash

### Attach a log file

[RPCS3.log](https://github.com/RPCS3/rpcs3/files/9759731/RPCS3.log)

### System configuration

Steam Deck

**Comment by Vestrel** [comment]
Try with #12805

**Comment by Vestrel** [comment]
Use build from the linked pr, not master

**Comment by Skyrimer3d** [comment]
I also have this issue, but as a Windows guy using Steam Deck and Linux for the first time, I'm clueless about how to test https://github.com/RPCS3/rpcs3/pull/12805

**Comment by Vestrel** [comment]
Download artifact from this link https://cirrus-ci.com/task/6563592441954304

**Comment by marcoluc97** [comment]
Retry latest version and let us know that PR is now merged. 
@Skyrimer3d @cmalvi

**Comment by cmalvi** [comment]
> Retry latest version and let us know that PR is now merged. 
> 
> @Skyrimer3d @cmalvi 

After a quick test here's my feedback:

- launch a game without Bluetooth headphones

- connect Bluetooth headphones, no more crash

- audio now switch output to headphones correctly

- put headphones in their case to disconnect

- rpcs3 now doesn't output audio to speaker 

- reconnect headphones

- now rpcs3 output to speaker again 

- disconnect headphones

- rpcs3 mute again

Note: loop last 5 step

So it works but there are some problems returning to previous output

**Comment by Megamouse** [comment]
So there must be an issue with the device choice after a disconnect.

**Comment by Vestrel** [comment]
Post new log

**Comment by cmalvi** [comment]
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/9843771/RPCS3.log)

**Comment by Skyrimer3d** [comment]
Downloaded the latest update from Steam Deck using Discover, and this seems fixed to me. What I did:
1. Booted game, sounds is coming through speakers.
2. Connected to bluetooth speakers (google nest mini, a surprisingly decent BT speaker), connects fine, no crashing, audio is ok.
3. Turned SD to sleep
4. rebooted, audio is coming from speakers. Connected to Bluetooth speakers again , no issues.

Repeated this process twice with Spiderman Shattered Dimensions and still working fine, so it looks fixed to me.

**Comment by cmalvi** [comment]
> Downloaded the latest update from Steam Deck using Discover, and this seems fixed to me. What I did:
> 
> 1. Booted game, sounds is coming through speakers.
> 
> 2. Connected to bluetooth speakers (google nest mini, a surprisingly decent BT speaker), connects fine, no crashing, audio is ok.
> 
> 3. Turned SD to sleep
> 
> 4. rebooted, audio is coming from speakers. Connected to Bluetooth speakers again , no issues.
> 
> 
> 
> Repeated this process twice with Spiderman Shattered Dimensions and still working fine, so it looks fixed to me.

Flatpak version isn't an official build

**Comment by digant73** [comment]
@cmalvi is that still an issue?

**Comment by cmalvi** [comment]
No, only one time it refuse to switch output but doesn't crash anymore

**Comment by digant73** [comment]
ok. if the issue is fixed please close this ricket
